src/components/TodoField/TodoField.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { updateTodo } from '../../api/todos';
+import { DispatchContext } from '../../State/State';
+import { handleDeleteTodo } from '../../services/todoItemServices';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoField: React.FC<Props> = ({ todo }) => {
+  const { title, id } = todo;
+  const [currentTitle, setCurrentTitle] = useState(title);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const edit = useRef<HTMLInputElement>(null);
+  const dispatch = useContext(DispatchContext);
+
+  useEffect(() => {
+    if (edit.current) {
+      edit.current.focus();
+    }
+  }, [isEditing]);
+
+  function editTodo(event: React.FormEvent) {
+    event.preventDefault();
+    dispatch({ type: 'setIsSubmitting', payload: true });
+
+    const promise: Promise<void> = new Promise((resolve) => {
```

Comment:
Why did you create a separate new promise here? just call `updateTodo` inside if